### PR TITLE
[FSDE-806][FSDE-807][FSDE-808] Add support of asynchronous nested row content, default sort and hide on single page configs

### PR DESCRIPTION
**Pull request by janibolkvadze**
### Commit summary

- [x] Added support of asynchronous updating of row nested content
- [x] Added support of skeleton template for loading state
- [x] Added support of `defaultSort` configuration to configure automatic sorting by the specific column and direction.
- [x] Added support of `chiRowExpanded`, `chiRowCollapsed` event emission
- [x] Fixed lowercase-uppercase sorting issue
- [x] Added support of `hideOnSinglePage` to configure hiding of the Pagination in single-page scenario
- [x] Changed Data Table property names with leading `_` underscore to avoid initial double-rendering (using leading underscore non-reactiveness of Vue)

@dani-cl-madrid @SergioQCL @jllr @mattnickles 

https://ctl.atlassian.net/browse/FSDE-806
https://ctl.atlassian.net/browse/FSDE-807
https://ctl.atlassian.net/browse/FSDE-808

### CODE REVIEW

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-66/1/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-66/2/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-66/3/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-66/4/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://d15c06185730bb5532e7067cc86588455bbd5a6a-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-66/6/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://91d0b3304469e2ea7598c413b834310d516045cb-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-66/7/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://8eb1889e0dcd5410e30044e386d00e559007a12e-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-66/8/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://83858441f65070a425e8893f50c0757b97d2aef5-chi-pr.enterprise.ctl.io

File: src/chi-vue/src/views/DataTableView.vue [hunk 1]
```diff
@@ -40,6 +42,14 @@
           <ChiDataTableFilters :filtersData="toolbar.filtersData" class="-ml--2" />
         </ChiDataTableToolbar>
       </template>
+      <template #loadingSkeleton>
+        <div class="-d--flex -flex--column -w--100">
+          <div style="width: 420px;" class="chi-skeleton -mw--100"></div>
+          <div style="width: 960px;" class="chi-skeleton -xs -mw--100 -mt--2"></div>
+          <div style="width: 1024px;" class="chi-skeleton -xs -mw--100 -mt--1"></div>
+          <div style="width: 720px;" class="chi-skeleton -xs -mw--100 -mt--1"></div>
```
Comment (SergioQCL) [inline src/chi-vue/src/views/DataTableView.vue:50]: Question: aren't there any utility classes which can replace these inline styles? Example: -w--960. If not..does it make sense to include them using some predefined sizes like -mw does?

File: src/chi-vue/src/components/data-table/DataTable.tsx [hunk 2]
```diff
@@ -187,9 +203,12 @@ export default class DataTable extends Vue {
     });
   }
 
-  cellAlignment(align: string): string {
-    // eslint-disable-next-line
-    const alignmentUtilityClasses: any = {
+  _cellAlignment(align: 'left' | 'center' | 'right'): string {
```
Comment (SergioQCL) [inline src/chi-vue/src/components/data-table/DataTable.tsx:206]: Will it make sense some alignType?

File: src/chi-vue/src/views/DataTableView.vue [hunk 3]
```diff
@@ -81,6 +92,12 @@ import DataTableFilters from '../components/data-table-filters/DataTableFilters'
     filtersChange: e => {
       console.log(e);
     },
+    rowExpanded: e => {
+      console.log(`Expanded row: `, e);
+    },
+    rowCollapsed: e => {
+      console.log(`Collapsed row: `, e);
+    },
```
Comment (dani-cl-madrid) [inline src/chi-vue/src/views/DataTableView.vue:100]: We should remove these messages for production. If you'd like to enable logs in production, there should be a special way, but, by default, no logs should be shown in production.

**Review by dani-cl-madrid — CHANGES_REQUESTED** [review]
A tiny change.

File: src/chi-vue/src/views/DataTableView.vue (hunk 3 above)
Comment (janibolkvadze) [reply src/chi-vue/src/views/DataTableView.vue:100]: `DataTableView.vue` is not distributed in production it is a testing View.

These are the distributed components:

```
const exportComponent: ComponentLibrary = {
  components: {
    ChiSearchInput: () => import(/* webpackChunkName: "search-input" */ '../components/search-input/SearchInput'),
    ChiDataTable: () => import(/* webpackChunkName: "data-table" */ '../components/data-table/DataTable'),
    ChiDataTableToolbar: () =>
      import(/* webpackChunkName: "data-table-toolbar" */ '../components/data-table-toolbar/DataTableToolbar'),
    ChiDataTableFilters: () =>
      import(/* webpackChunkName: "data-table-filters" */ '../components/data-table-filters/DataTableFilters'),
    ChiDrawer: () => import(/* webpackChunkName: "drawer" */ '../components/drawer/drawer'),
    ChiPagination: () => import(/* webpackChunkName: "pagination" */ '../components/pagination/pagination'),
    ChiTooltip: () => import(/* webpackChunkName: "tooltip" */ '../components/tooltip/tooltip'),
  },
};
```

**Review by dani-cl-madrid — APPROVED** [review]

File: src/chi-vue/src/views/DataTableView.vue (hunk 1 above)
Comment (janibolkvadze) [reply src/chi-vue/src/views/DataTableView.vue:50]: We do have max-width sizing utility classes like `-mw--480` but not width ones like `-w--480`.
@jllr Thoughts on the suggestion of adding this kind of adding width utility classes with predefined sizes?

File: src/chi-vue/src/components/data-table/DataTable.tsx (hunk 2 above)
Comment (janibolkvadze) [reply src/chi-vue/src/components/data-table/DataTable.tsx:206]: Sounds good 🤝

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-66/9/. :x:

File: src/chi-vue/src/views/DataTableView.vue (hunk 1 above)
Comment (janibolkvadze) [reply src/chi-vue/src/views/DataTableView.vue]: Replaced the inline styles with -w--* utility classes and made them responsive to screen size. Thanks for the suggestion!

**Review by SergioQCL — APPROVED** [review]
LGTM

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-66/10/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-66/11/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-66/12/. :x:

**Review by jllr — APPROVED** [review]

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-66/13/. :x:

**Pull request closed without merging**